enabler/src/de/schildbach/pte/NegentweeProvider.java

```diff
@@ -0,0 +1,986 @@
+package de.schildbach.pte;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Currency;
+import java.util.Date;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.text.SimpleDateFormat;
+
+import javax.annotation.Nullable;
+
+import de.schildbach.pte.dto.Departure;
+import de.schildbach.pte.dto.Fare;
+import de.schildbach.pte.dto.Line;
+import de.schildbach.pte.dto.LineDestination;
+import de.schildbach.pte.dto.Location;
+import de.schildbach.pte.dto.LocationType;
+import de.schildbach.pte.dto.NearbyLocationsResult;
+import de.schildbach.pte.dto.Point;
+import de.schildbach.pte.dto.Position;
+import de.schildbach.pte.dto.Product;
+import de.schildbach.pte.dto.QueryDeparturesResult;
+import de.schildbach.pte.dto.QueryTripsContext;
+import de.schildbach.pte.dto.QueryTripsResult;
+import de.schildbach.pte.dto.ResultHeader;
+import de.schildbach.pte.dto.StationDepartures;
+import de.schildbach.pte.dto.Stop;
+import de.schildbach.pte.dto.SuggestLocationsResult;
+
+import de.schildbach.pte.dto.SuggestedLocation;
+import de.schildbach.pte.dto.Trip;
+import de.schildbach.pte.exception.InternalErrorException;
+import de.schildbach.pte.exception.NotFoundException;
+import de.schildbach.pte.util.ParserUtils;
+import de.schildbach.pte.util.WordUtils;
+import okhttp3.HttpUrl;
+
+
+public class NegentweeProvider extends AbstractNetworkProvider {
+
+    private static final String API_BASE = "https://api.9292.nl/0.1/";
+    private static final String SERVER_PRODUCT = "negentwee";
+
+    private static final Language DEFAULT_API_LANG = Language.NL_NL;
+    private static final int DEFAULT_MAX_LOCATIONS = 50;
+
+    private static final SimpleDateFormat dateTimeParser = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm");
+    private static final SimpleDateFormat timeParser = new SimpleDateFormat("HH:mm");
+
+    private final Language language;
+    private final ResultHeader resultHeader;
+
+    public enum Language {
+        NL_NL("nl-NL"),
+        EN_GB("en-GB");
+
+        private final String lang;
+
+        Language(String lang) {
+            this.lang = lang;
+        }
+
+        public String toString() {
+            return this.lang;
+        }
+    }
+
+    private enum InterchangeTime {
+        STANDARD,
+        EXTRA;
+
+        public String toString() {
+            return name().toLowerCase();
+        }
+    }
+
+    public static class QueryParameter implements Serializable {
+        public String name, value;
+
+        QueryParameter(String name, String value){
```

Comment:
This class is an implementation detail only used internally that I made public so I could use it in a test during development. I made the whole class private instead (if you want to reuse it somewhere and make it public, it should probably be moved to another file).